src/DomCrawler/Crawler.php

```diff
@@ -86,7 +86,7 @@ public function addNode(\DOMNode $node): void
         throw $this->createNotSupportedException(__METHOD__);
     }
 
-    public function eq($position): self
+    public function eq($position): static
```

Comment:
This is another spot where we can't do this... unless we're planning to drop support for PHP 7. But... maybe we are??? In order to support Symfony 6?

Reply:
I guess the core team of Symfony should take a decision on this. I can work on both implementations. Keeping support for older php version needs main Symfony repository to be adapted before 6.0 comes out. Removing support for older php versions might need a major version of this package.

Reply:
But indeed this one I missed when reporting here: https://github.com/symfony/symfony/issues/43021

This will be a ton of changes, because all of. this class changed its return types. 😱

Reply:
Either we remove the return types from core, or we add some conditional classes/traits declarations in this bundle. 
I'm personally ok to revert the addition of these type declarations, we can re-add them later.

Reply:
So, should we revert these in core? If we're going to do that, I guess we should do it as quickly as we can.

(I'm still very unfamiliar with the typing situation - so apologies for not having anything intelligent to add)

Reply:
Yes, I am trying to fix everything here to know what types exactly should be reverted, because I wasn't taking care of static.

My idea is:
* Prepare this PR to be able to ensure that changes on Symfony will make this PR pass
* Make a PR on Symfony core to change all types needed here
* Adapt this PR when the PR on the main repo is merged

For the moment I am having some trouble with PHPStan, since it is not a dependency of this project it got upgraded to 1.0 without notice and now it reports more errors.

Reply:
This makes sense to me - thank you and keep us posted :)

Reply:
Im trying a little trick to patch symfony to remove its return types so we can ensure that those changes applied to Symfony 6.0 will make build pass with all versions here.

So far I think it will work, but I am having some last troubles with PHPStan.

The idea is to remove the last 2 commits from this PR once I do the Symfony PR to remove the return typehints

wdyt?